### ISSUE TITLE
Console errors since upgrade to v1.0.975

### ISSUE THREAD

**Issue by johannesjo** [wallabyjs/public#2538]
Since the newest version (installed today on the 14th of October 2020, v1.0.975) I get errors in the wallabyconsole for IntelliJ Idea 2020.2.3.

```
wallaby.js started
core v1.0.975
Warning: Entry point 'SECRET' contains deep imports into '/media/onono/CANTTELL/node_modules/SECRET/ba'. This is probably not a problem, but may cause the compilation of entry points to be out of order.
Warning: Entry point 'SECRET/ui' contains deep imports into '/media/onono/CANTTELL/node_modules/dayjs/plugin/weekOfYear'. This is probably not a problem, but may cause the compilation of entry points to be out of order.
Runtime error: Test file xyz.component.spec.ts is being imported from other test files.
This may cause tests in the file to be executed more than once (when the test file runs and every time it gets imported).
If the test file contains some test helper code, try refactoring the reused code into a file of its own and use it by importing the new helper file where required.
    at Object.window.__moduleBundler.assert (http://localhost:40157/____wallaby-webpack-bootstrapper.js?1602678157803:8:17)
    at window.__moduleBundler.require (http://localhost:40157/____wallaby-webpack.js?1602678157803:12:30)
    at Module.window.__moduleBundler.modules.<computed> (http://localhost:40157/SCRET/src/app/some.ts.wbp.js?1602678157624&wallabyFileId=137:7:77)
    at Object.window.__moduleBundler.require (http://localhost:40157/____wallaby-webpack.js?1602678157803:28:16)
    at http://localhost:40157/____wallaby-webpack-bootstrapper.js?1602678157803:18:36
    at http://localhost:40157/__wallaby__/tracer.js:15:11323
    at u._doWhenReceiverIsReady (http://localhost:40157/__wallaby__/tracer.js:14:10070)
    at Object._startWhenReceiverIsReady (http://localhost:40157/__wallaby__/tracer.js:15:11289)
    at Object.window.__moduleBundler.loadTests (http://localhost:40157/____wallaby-webpack-bootstrapper.js?1602678157803:13:22)
    at http://localhost:40157/wallaby_sandbox1.html:9:67643
    at http://localhost:40157/wallaby_sandbox1.html:9:67657
Runtime error: Test file SCRET/src/app/modules/SCRET/secret.component.spec.ts is being imported from other test files.
This may cause tests in the file to be executed more than once (when the test file runs and every time it gets imported).
If the test file contains some test helper code, try refactoring the reused code into a file of its own and use it by importing the new helper file where required.
    at Object.window.__moduleBundler.assert (http://localhost:40157/____wallaby-webpack-bootstrapper.js?1602678157803:8:17)
    at window.__moduleBundler.require (http://localhost:40157/____wallaby-webpack.js?1602678157803:12:30)
    at Module.window.__moduleBundler.modules.<computed> (http://localhost:40157/SCRET/src/app/NONO.spec.ts.wbp.js?1602678157613&wallabyFileId=100:10:77)
    at Object.window.__moduleBundler.require (http://localhost:40157/____wallaby-webpack.js?1602678157803:28:16)
    at http://localhost:40157/____wallaby-webpack-bootstrapper.js?1602678157803:18:36
    at http://localhost:40157/__wallaby__/tracer.js:15:11323
    at u._doWhenReceiverIsReady (http://localhost:40157/__wallaby__/tracer.js:14:10070)
    at Object._startWhenReceiverIsReady (http://localhost:40157/__wallaby__/tracer.js:15:11289)
    at Object.window.__moduleBundler.loadTests (http://localhost:40157/____wallaby-webpack-bootstrapper.js?1602678157803:13:22)
    at http://localhost:40157/wallaby_sandbox0.html:9:67661
    at http://localhost:40157/wallaby_sandbox0.html:9:67675
Execution progress: 50 tests
Execution progress: 100 tests
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/3915fb6d-bb50-40b2-b88a-af494d9c1742:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/3915fb6d-bb50-40b2-b88a-af494d9c1742:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/b4046c14-0917-4d5b-81a8-8c7b4362aabb:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/b4046c14-0917-4d5b-81a8-8c7b4362aabb:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/91c185b7-daae-44f6-88a0-c7b0b232ec0f:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/91c185b7-daae-44f6-88a0-c7b0b232ec0f:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/d913dc34-ad34-4ee6-b6a3-2d97302cafe0:2
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:40157/d913dc34-ad34-4ee6-b6a3-2d97302cafe0:2
Finished executing 103 affected test(s) in 10.17 second(s)

```

**Comment by Dor256** [comment]
I'm also experiencing issues with v1.0.975...
When trying to run the tests I'm getting the following output from the wallaby.js console:
```‍
[Info]  Started Wallaby.js Core v1.0.975
[Error] Failed to load configuration file: Automatic Jest configuration error: Cannot read property 'module' of undefined
```

Weirdly enough, for some reason only one project of mine runs on v1.0.975 and crashes this way.
The rest of my projects use v1.0.974 and I experience no issues with the exact same configuration.

Any help would be much appreciated!

**Comment by smcenlly** [comment]
@johannesjo:
Can you please provide your [Wallaby diagnostics report](https://wallabyjs.com/docs/intro/troubleshooting.html#diagnostics-report) so we can see your project configuration and some additional log details. To share the report:
- start Wallaby in your project;
- wait until it reports some results (finishes running tests);
- run `Copy Diagnostics Report` command:
  - in VS Code, Atom, or Sublime editor execute Wallaby `Copy Diagnostics Report` command from your editor command palette; 
  - in JetBrains IDEs or VS use Wallaby `Help | Wallaby.js | Copy Diagnostics Report` menu item;
- paste generated report from your clipboard in your reply (feel free to review and edit the report if required).

You also have a runtime error reported by Wallaby `Test file xyz.component.spec.ts is being imported from other test files.` Is this intentional? If so, I'm assuming that this spec file is not an entry point to your test and you may want to exclude it from your test files, or else fix where it is imported?

@Dor256:
Can you please confirm that `jest` is working from the command line (e.g. using `npx jest`)? If not, please fix this first and then try Wallaby again.

It is possible that there is a problem with your `node_modules`, please try deleting your `node_modules` and re-installing your project dependencies.

If you're still having problems after re-installing your project dependencies, can you please [create a new issue](https://github.com/wallabyjs/public/issues/new) and include your `package.json` dependencies and devDependencies.

**Comment by Dor256** [comment]
@smcenlly 
Hey, thanks for the reply,
`jest` is working fine and the problem persists after deleting `node_modules` and reinstalling. I'll open a new issue with all of the details, 

Thanks!

**Comment by johannesjo** [comment]
@smcenlly can I send you the data per mail? I cleaned it up as good as I can, but my employer is very very strict about this.

**Comment by ArtemGovorov** [comment]
@johannesjo sure, please send the email to hello@wallabyjs.com

**Comment by johannesjo** [comment]
Sent. Thank you!

**Comment by smcenlly** [comment]
Thanks for sending through your diagnostics report. As an aside, if you wish to conduct our interactions entirely over email because of concerns sharing information publicly, feel free to send your reply via email again to hello@wallabyjs.com and we can manage our response there.

Your diagnostic report has an error:

```
Uncaught Error: Test file <NAME_REMOVED> is being imported from other test files.
This may cause tests in the file to be executed more than once (when the test file runs and every time it gets imported).
If the test file contains some test helper code, try refactoring the reused code into a file of its own and use it by importing the new helper file where required.
```

Is the file referenced in the error being imported from other test files? We're assuming from the error message that it is and expect that this is the result of a recent change?

Wallaby does not support importing test files from other test files. As per the error message, `If the test file contains some test helper code, try refactoring the reused code into a file of its own and use it by importing the new helper file where required.`. If the file being imported does not have tests (e.g. is only helper/util code), the problem is that the file name matches your test patterns so renaming the file should fix your problem.

---

We also could not see the error reported above as part of your diagnostics report:

```
Runtime error: Uncaught ReferenceError: $_$wf is not defined
```

Are you still getting that particular error? If so, please fix the test file import first as this may be related to the first problem.

**Comment by johannesjo** [comment]
Even with adjusting the imports, there are still 8 out of 133 tests failing, even though they are not when running tests normally from the console. They all fail with the following error:
```

	Uncaught ReferenceError: $_$wf is not defined

	Uncaught ReferenceError: $_$wf is not defined
```

**Comment by johannesjo** [comment]
As I said this was working with the previous version. 

Here is some excerpt from the diagnostics report:
```
     '2020-10-19T12:05:07.790Z fs File changed in editor: XXXX.ts\n',
     '2020-10-19T12:05:07.799Z extended-core File was not changed, but file markers may need to be synced\n',
     '2020-10-19T12:05:07.844Z project Running postprocessor\n',
     '2020-10-19T12:05:07.844Z testTask Test files from affected: 2, from deleted or manually requested: 0, from recently changed: 0, from loaded by: 4, from failing: 1\n',
     '2020-10-19T12:05:07.845Z postprocessor Processing started.\n',
     '2020-10-19T12:05:07.845Z postprocessor Compilation started (incremental).\n',
     '2020-10-19T12:05:08.583Z postprocessor Compilation finished.\n',
     '2020-10-19T12:05:08.800Z postprocessor Emitting 12 files.\n',
     '2020-10-19T12:05:08.806Z postprocessor Processing finished.\n',
     '2020-10-19T12:05:08.806Z project Postprocessor execution finished\n',
     '2020-10-19T12:05:08.806Z project Test run started; run priority: 2\n',
     '2020-10-19T12:05:08.808Z testTask Test files from affected: 2, from deleted or manually requested: 0, from recently changed: 0, from loaded by: 4, from failing: 1\n',
     '2020-10-19T12:05:08.817Z workers Starting test run, priority: 2\n',
     '2020-10-19T12:05:08.817Z workers Distributing tests between 5 workers\n',
     '2020-10-19T12:05:08.817Z workers Running tests in parallel\n',
     '2020-10-19T12:05:08.817Z chromeRunner Starting sandbox [worker #0, session #40xen]\n',
     '2020-10-19T12:05:08.817Z chromeRunner Starting sandbox [worker #1, session #bf8dr]\n',
     '2020-10-19T12:05:08.817Z chromeRunner Starting sandbox [worker #2, session #l0asm]\n',
     '2020-10-19T12:05:08.817Z chromeRunner Starting sandbox [worker #3, session #5imep]\n',
     '2020-10-19T12:05:08.817Z chromeRunner Starting sandbox [worker #4, session #0xfay]\n',
     '2020-10-19T12:05:08.817Z chromeRunner Preparing sandbox [worker #0, session #40xen]\n',
     '2020-10-19T12:05:08.818Z browserRunner Total files to load in sandbox: 14\n',
     '2020-10-19T12:05:08.820Z browserRunner Sandbox is generated [worker #0, session #40xen]: http://localhost:43809/wallaby_sandbox0.html\n',
     '2020-10-19T12:05:08.820Z chromeRunner Preparing sandbox [worker #1, session #bf8dr]\n',
     '2020-10-19T12:05:08.820Z browserRunner Total files to load in sandbox: 15\n',
     '2020-10-19T12:05:08.822Z browserRunner Sandbox is generated [worker #1, session #bf8dr]: http://localhost:43809/wallaby_sandbox1.html\n',
     '2020-10-19T12:05:08.822Z chromeRunner Preparing sandbox [worker #2, session #l0asm]\n',
     '2020-10-19T12:05:08.822Z browserRunner Total files to load in sandbox: 14\n',
     '2020-10-19T12:05:08.823Z browserRunner Sandbox is generated [worker #2, session #l0asm]: http://localhost:43809/wallaby_sandbox2.html\n',
     '2020-10-19T12:05:08.823Z chromeRunner Preparing sandbox [worker #3, session #5imep]\n',
     '2020-10-19T12:05:08.823Z browserRunner Total files to load in sandbox: 14\n',
     '2020-10-19T12:05:08.824Z browserRunner Sandbox is generated [worker #3, session #5imep]: http://localhost:43809/wallaby_sandbox3.html\n',
     '2020-10-19T12:05:08.824Z chromeRunner Preparing sandbox [worker #4, session #0xfay]\n',
     '2020-10-19T12:05:08.829Z browserRunner Total files to load in sandbox: 14\n',
     '2020-10-19T12:05:08.830Z browserRunner Sandbox is generated [worker #4, session #0xfay]: http://localhost:43809/wallaby_sandbox4.html\n',
     '2020-10-19T12:05:08.830Z chromeRunner Prepared sandbox [worker #0, session #40xen]\n',
     '2020-10-19T12:05:08.830Z chromeRunner Prepared sandbox [worker #1, session #bf8dr]\n',
     '2020-10-19T12:05:08.830Z chromeRunner Prepared sandbox [worker #2, session #l0asm]\n',
     '2020-10-19T12:05:08.830Z chromeRunner Prepared sandbox [worker #3, session #5imep]\n',
     '2020-10-19T12:05:08.830Z chromeRunner Prepared sandbox [worker #4, session #0xfay]\n',
     '2020-10-19T12:05:08.830Z workers [worker #0, session #40xen] Running tests in sandbox\n',
     '2020-10-19T12:05:08.831Z workers [worker #1, session #bf8dr] Running tests in sandbox\n',
     '2020-10-19T12:05:08.831Z workers [worker #2, session #l0asm] Running tests in sandbox\n',
     '2020-10-19T12:05:08.831Z workers [worker #3, session #5imep] Running tests in sandbox\n',
     '2020-10-19T12:05:08.831Z workers [worker #4, session #0xfay] Running tests in sandbox\n',
     '2020-10-19T12:05:09.504Z workers [bf8dr] Loaded 47 test(s)\n',
     '2020-10-19T12:05:09.514Z workers [40xen] Loaded 4 test(s)\n',
     '2020-10-19T12:05:09.518Z workers [5imep] Loaded 4 test(s)\n',
     '2020-10-19T12:05:09.520Z workers [0xfay] Loaded 2 test(s)\n',
     '2020-10-19T12:05:09.535Z workers [l0asm] Loaded 7 test(s)\n',
     '2020-10-19T12:05:09.543Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.575Z workers [0xfay] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.585Z workers [5imep] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.591Z workers [40xen] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.593Z workers [0xfay] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.615Z workers [5imep] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.628Z workers [40xen] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.631Z workers [0xfay] Run 2 test(s), skipped 0 test(s)\n',
     '2020-10-19T12:05:09.632Z workers [0xfay] Sandbox is responsive, closing it\n',
     '2020-10-19T12:05:09.641Z workers [5imep] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.665Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.667Z workers [5imep] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.702Z workers [40xen] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.709Z workers [5imep] Run 4 test(s), skipped 0 test(s)\n',
     '2020-10-19T12:05:09.710Z workers [5imep] Sandbox is responsive, closing it\n',
     '2020-10-19T12:05:09.721Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.726Z workers [40xen] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.756Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.763Z workers [40xen] Run 4 test(s), skipped 0 test(s)\n',
     '2020-10-19T12:05:09.766Z workers [40xen] Sandbox is responsive, closing it\n',
     '2020-10-19T12:05:09.776Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.798Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.829Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.859Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.885Z workers [l0asm] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:09.927Z workers [l0asm] Run 7 test(s), skipped 0 test(s)\n',
     '2020-10-19T12:05:09.928Z workers [l0asm] Sandbox is responsive, closing it\n',
     '2020-10-19T12:05:10.466Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:11.041Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:11.650Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:11.854Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:11.854Z workers Failed to map the stack to user code, entry message: Uncaught ReferenceError: $_$wf is not defined, stack: Uncaught ReferenceError: $_$wf is not defined\nat blob:http://localhost:43809/047334ad-4f70-4fe2-a578-72f361e15148:2\n',
     '2020-10-19T12:05:12.839Z workers [bf8dr] Test executed: should do something xyz different tests\n',
     '2020-10-19T12:05:12.839Z workers Failed to map the stack to user code, entry message: Uncaught ReferenceError: $_$wf is not defined, stack: Uncaught ReferenceError: $_$wf is not defined\nat blob:http://localhost:43809/2318a95e-c74f-4783-968b-e6367a537964:2\n',
     '2020-10-19T12:05:12.839Z workers Failed to map the stack to user code, entry message: Uncaught ReferenceError: $_$wf is not defined, stack: Uncaught ReferenceError: $_$wf is not defined\nat blob:http://localhost:43809/2318a95e-c74f-4783-968b-e6367a537964:2\n',
     '2020-10-19T12:05:12.964Z workers Sandbox (active) [bf8dr] error: Uncaught ReferenceError: $_$wf is not defined\n',
     '2020-10-19T12:05:12.964Z workers Failed to map the stack to user code, entry message: Uncaught ReferenceError: $_$wf is not defined, stack: Uncaught ReferenceError: $_$wf is not defined\nat blob:http://localhost:43809/047334ad-4f70-4fe2-a578-72f361e15148:2\n',
     '2020-10-19T12:05:13.064Z workers [bf8dr] Run 7 test(s), skipped 0 test(s)\n',
     '2020-10-19T12:05:13.353Z workers [bf8dr] Sandbox is responsive, closing it\n',
     '2020-10-19T12:05:13.354Z workers Merging parallel test run results\n',
     '2020-10-19T12:05:13.375Z project Test run finished\n',
     '2020-10-19T12:05:13.375Z project Processed console.log entries\n',
     '2020-10-19T12:05:13.375Z project Processed loading sequences\n',
     '2020-10-19T12:05:13.375Z project Processed executed tests\n',
     '2020-10-19T12:05:13.410Z project Processed code coverage\n',
     '2020-10-19T12:05:13.477Z project Test run result processed and sent to IDE\n',
     '2020-10-19T12:45:06.348Z uiService UI client connected\n',
     '2020-10-19T12:45:06.349Z uiService Outgoing message ui:handshake\n',
     '2020-10-19T12:45:06.395Z uiService Incoming message ui:start\n',
     '2020-10-19T12:45:06.395Z uiService Outgoing message ui:summary\n',
     '2020-10-19T12:45:06.416Z uiService Outgoing message ui:files\n',
     '2020-10-19T12:45:06.417Z uiService Incoming message ui:tests:resultsRequested\n',
     '2020-10-19T12:45:06.430Z uiService Outgoing message ui:tests:allResultsUpdated\n' ] }
```

**Comment by smcenlly** [comment]
> As I said this was working with the previous version.

We've reviewed our recent changes and there does not appear to be anything that would could see that could cause your problem after upgrading from a fairly recent version to v1.0.975. If it had been a while since you used Wallaby, then it's much more possible there was some change that is causing your problem but we still suspect it's project related. If you are interested in checking a previous version, you may follow our [offline installation instructions](https://wallabyjs.com/docs/intro/offline.html#vs-code) and when updating the `core version`, download a previous version from `https://wallaby-downloads.s3.amazonaws.com/wallaby-vNNN.zip`, where `NNN` is the previous version (e.g. https://wallaby-downloads.s3.amazonaws.com/wallaby-v1.0.973.zip).

> Even with adjusting the imports, there are still 8 out of 133 tests failing, even though they are not when running tests normally from the console. They all fail with the following error:
> ```
> Uncaught ReferenceError: $_$wf is not defined
> ```
> 

When Wallaby runs your tests and code, it modifies your code before executing it and adds some extra code in places (without modifying your program's logic) to determine code coverage, live value display, etc. The error that you are receiving indicates that one of the files that is instrumented by Wallaby is being executed outside of a Wallaby context, or before Wallaby is initialized. This is pretty rare but can be caused for example by Wallaby instrumenting a file that is used in your global setup (outside of the context of running your tests). When your code runs on this file, Wallaby is not running or started and so you get the error that you're seeing. You may see the same behavior if for example you had code that spawned another process to perform some initialization logic in your tests. 

You should be able to see determine what code files are causing a problem by navigating to the URL provided in your console log errors. For example, for:

```
     '2020-10-19T12:05:11.854Z workers Failed to map the stack to user code, entry message: Uncaught ReferenceError: $_$wf is not defined, stack: Uncaught ReferenceError: $_$wf is not defined\nat blob:http://localhost:43809/047334ad-4f70-4fe2-a578-72f361e15148:2\n',
```

Open `http://localhost:43809/047334ad-4f70-4fe2-a578-72f361e15148` in your browser and you should be able to determine what imported code is causing the problem and why. Depending on what is happening and why, you may be able to use [Wallaby configuration overrides](https://wallabyjs.com/docs/integration/angular.html#automatic-configuration-workspace-overrides) for your Angular project(s) to address the problem. You may also override Wallaby's automatic configuration defaults to excludes the files that are problematic from being instrumented by creating a **Wallaby configuration file** or by updating your **package.json** (see below). From your logs, it looks like you have two files that you need to stop Wallaby from instrumenting.

**wallaby.js configuration file**
```javascript
module.exports = () => ({
  autoDetect: true,
  files: [{
    pattern: 'relative/path/to/file.ts', instrument: false
  }]
});
```

**package.json**
```javascript
{
  "name": "...",
  ...

  "dependencies": {
    ...
  },

  "devDependencies": {
    ...
  },

  ...

  "wallaby": {
     "files": [
      { "pattern": "relative/path/to/file.ts", "instrument": false }
    ]
 }
}
```

**Comment by ArtemGovorov** [comment]
> there are still 8 out of 133 tests failing, even though they are not when running tests normally from the console. 

Do these tests have anything in common? As @smcenlly mentioned, the error that you are receiving indicates that one of the files that is instrumented by Wallaby, specifically one of the functions of the file, is being executed outside of a Wallaby context. Maybe there's something special that the 8 tests are doing, such as passing functions to be executed elsewhere outside of your web page context?

**Comment by johannesjo** [comment]
You might want to update your docs with the correct path for Linux `.cache/JetBrains/xxx`.

**Comment by johannesjo** [comment]
Just tested with  v1.0.973. The coverage is a bit broken, but the tests don't fail. I also experimented a bit. For the newest version, it seems that the test are failing somewhat randomly. If I comment out a couple of tests, others seem to fail. I cannot reproduce this behaviour for v1.0.973.

**Comment by johannesjo** [comment]
If I open `http://localhost:43809/047334ad-4f70-4fe2-a578-72f361e15148` (or similar) in my browser  I get a 404.

**Comment by johannesjo** [comment]
> Do these tests have anything in common?

These are all tests of the same service, inside the same testing file.

**Comment by smcenlly** [comment]
> You might want to update your docs with the correct path for Linux .cache/JetBrains/xxx. Also the wallaby/wallaby-app path is wallaby/wallaby` actually.

Thanks for letting us know - I assume you're referring our offline installation docs.

> Just tested with v1.0.973. The coverage is a bit broken, but the tests don't fail. 

Can you please provide some details on what you're seeing? How is the coverage broken, do you have any errors reported in your Wallaby console/diagnostics report?

> For the newest version, it seems that the test are failing somewhat randomly.

You mention the tests are failing somewhat randomly; do they always fail? If so I assume they're failing in different ways/places? Do you get the same error message each time or are the failure reasons different?

There are no changes between version `v1.0.973` and `v1.0.975` that would make Wallaby run differently in each version. We expect that the difference is either project related or something environmental. We suspect that the same issue exists in both versions but exhibit as broken coverage in one version and failing tests in the other due because of project/environmental reasons. Can you also confirm that you used the same project folder for both versions, or did you clone the project to try the older version? Could you also please try cloning your project to a new folder to see if that fixes your problem.

One possible cause of your problem is that Wallaby will split your test across processes and execute them in parallel. If you have multiple files that depend on the same code or the order of tests is important, this can lead Wallaby to fail randomly. To try mitigate this (at least for the initial execution of your tests at Wallaby startup), you can try creating a Wallaby configuration file and selecting that for your project to specify the workers section that limits Wallaby to use only one process (see below). If you do this, make sure that your IntelliJ IDEA Wallaby Runtime Configuration is updated to point to the configuration file. You may also update these settings in your `package.json` (no configuration file or Runtime Configuration changes required).

```javascript
module.exports = () => ({
  autoDetect: true,
  workers: {
       initial: 1,
       regular: 1
     }
});
```

---

To try and identify the cause of the problem, we need you to isolate the issue to a single test or minimum set of tests that is causing the problem. You may force wallaby to select and run this specific test file only, by adding the special `file.only` comment anywhere in the test file:

```javascript
// file.only
...
```

Once you have done this, you may further isolate the issue by starting to remove code from your test/source code until you have a minimum amount of code that is causing the problem. At this point you should have some generic details that you can provide to us so that we can attempt to reproduce your problem at our end.

Alternatively, could you provide a sample repo that has your problem? If you isolate the problem first, you should not need to provide any of your company-specific code. You may email this to us if you have any concerns around privacy.

**Comment by johannesjo** [comment]
I am not sure if I am reading this correctly, but the issue seems to be the import of a third-party library in the polyfills.ts  (not the tested file and not directly imported by it, but usually executed before that):
```
import './some-import';
```
Interestingly this leads the tests to sometimes work, but most of the time to fail with the `Uncaught ReferenceError: $_$wf is not defined` error. It's interesting that the third party provider stuff can be printed out correctly via console.log in the console of the actual application code. What is even more interesting is that if I log it out in the test, it doesn't fail anymore. Some code to illustrate:
```TypeScript
class MyService {
  thirdPartyFilledProp = new GlobalThirdPartySomething()
}
```
```TypeScript
it('test', ()=>{
  console.log(myService.thirdPartyFilledProp); // => doesn't fail any more, though this doesn't seem to be updated always...
});
```
```
// wallaby console output with console.log(myService.thirdPartyFilledProp)
Finished executing 4 affected test(s) in 2.84 second(s)

// wallaby console output WITHOUT console.log(myService.thirdPartyFilledProp)
Runtime error: Uncaught ReferenceError: $_$wf is not defined
at blob:http://localhost:38311/143d3723-ad9e-4978-b96c-baa0d4940d2b:2
Finished executing 1 affected test(s) in 1.38 second(s)


```

**Comment by smcenlly** [comment]
If you have a single test file with a simple test (e.g. `it('should work, () => {});`) do you still have the same problem?

Are you able to provide us with your `polyfills.ts` code so that we can try and replicate your problem ourselves?

**Comment by johannesjo** [comment]
> If you have a single test file with a simple test (e.g. `it('should work, () => {});`) do you still have the same problem?

You mean a single test file and nothing else? No, I don't. As explained above the error only occurs when I access the third party code.

```TypeScript

// some global ts file (bundled by webpack)
import './some-import';  // includes third party lib globally
// ---------

// other file
class MyService {
  thirdPartyFilledProp = new GlobalThirdPartySomething()
}

// test file => all fine
it('test', ()=>{
  console.log(myService.thirdPartyFilledProp); // => doesn't fail any more, though this doesn't seem to be updated always...
  myService.thirdPartyFilledProp.someMethod();
});


// test file  => broken
it('test', ()=>{
  myService.thirdPartyFilledProp.someMethod();
});
```

**Comment by smcenlly** [comment]
We have created and invited you to a private repo that has your project dependencies and angular project structure.

We tried adding `lodash` as a dependency and tested it from `asdasd/src/app/myService.spec.ts`. Everything seems to be working for us.

Can you please update the repo with the same scenario that you described above so that the project is breaking the same way and we can take a look?

**Comment by johannesjo** [comment]
You had me scared there for a moment :D I am glad to see that project dependency is limited to the angular stuff! As I said I have to be very secretive about the project, even far beyond the point where I personally think it makes sense...

I try to recreate the issue there. Thank you!

**Comment by johannesjo** [comment]
After literally hours of debugging and trying to create a valid test instance with the error, I give up for today. I managed to produce a new error, but that's not the source of my issue.  Maybe I give it another shot tomorrow.

> If I open http://localhost:43809/047334ad-4f70-4fe2-a578-72f361e15148 (or similar) in my browser I get a 404.

Do you have any idea why this is the case? Maybe this offers another angle on this.

**Comment by johannesjo** [comment]
I finally was able to make this reproducible. I pushed the code to the private repo. Please note that you have to trigger changes for the error to appear. Normally it works at first. After the first changed it should show the $wf error and afterward a new one. I also was able to pinpoint the exact commit when it stopped working but reverting that exact change does still not fix the $wf error so I am assuming there are multiple ways to cause it. It's safe to say, that wallaby does not play well with the here maps API stuff.

**Comment by smcenlly** [comment]
Thanks for updating the sample repo. We were able to fix the `$wf` problem.

We are still seeing some errors but believe that these are application related and not a problem with Wallaby running your tests. The errors appear to be caused by not having an API key to the third party library that you are using; we get the same errors using `ng test` (check your Chrome Developer Console). We also see some intermittent errors being reported by Wallaby due to a race condition between test execution finishing and async code in the third party library. If the third party library async code finishes (and errors) before your tests then Wallaby will report the error, otherwise Wallaby will not; this is expected behavior and something that could also affect your tests outside of Wallaby; we're guessing that this is also because we don't have an API key to the third party library.

We need you to create a wallaby.js configuration file and modify the files override function (see below) to exclude the 3rd party libraries from being instrumented by Wallaby:

**wallaby.js**
```javascript
module.exports = () => ({
  autoDetect: true,
  files: {
    override: files => {
      return [
        // list overrides at the start of the array (order is important)
        { pattern: 'project_name/src/third_party_scripts/**/*.*', instrument: false, load: false },

        // Existing file patterns are added at the end of the array
        ...files
      ];
    }
  }
});
```

After creating the configuration file, you'll need to update IntelliJ run configuration to use the configuration file instead of automatic configuration.

We're not sure why the third party library files are failing and because they're minified/obfuscated, we can't easily determine this. Regardless, you should not need Wallaby code-coverage on these files and I expect you don't want them to be processed by Wallaby.

---

A helpful tip that we wanted to share that we used when debugging is to update your environment configuration to run chrome in desktop mode and not close tabs after each run. This allows you to see any errors in your developer tools console the same as if you were running `ng test`. We don't recommend this mode of operation normally but sometimes it is helpful:

**wallaby.js configuration to show and keep chrome tabs open**
```javascript
const chromeFlags = ['--disable-gpu', '--deterministic-fetch'];
const os = require('os');
if (os.type() === 'Linux') {
  chromeFlags.push('--no-sandbox');
  chromeFlags.push('--disable-setuid-sandbox');
}

module.exports = () => ({
  autoDetect: true,
  env: {
    keepTabsOpened: true,
    kind: 'chrome',
    params: {
      runner: chromeFlags.join(' ')
    }
  }
});
```

**Comment by johannesjo** [comment]
Works like a charm! Thank you very much for your help!
